enhancements/hypershift/disaster-recovery/hcp-etcd-backup-api.md

```diff
@@ -0,0 +1,641 @@
+
+- [ETCD Operator Backup API Design Doc](https://docs.google.com/document/d/1xBnK3D7MCblFgxQ8RUZFyWNIU_bnSFHciLhhHAeP0Xk/edit)
+- [ETCD Backup API implementation](https://github.com/openshift/api/pull/1482)
+- [Periodic Backup Controller](https://github.com/openshift/cluster-etcd-operator/pull/1074)
+
+The other field is called `storeEndpoint` which basically will take care of the export of the backup object to an external endpoint. This endpoint could be S3, Azure, etc.... Inside of this `storeEndpoint` we have some fields:
+
+- **BackupPrefix:** Where the user could set some kind of prefix to identify that backup in a more readable way. This Prefix will be parsed, validated and used in the final name which will contain the **timestamp**, the **HostedCluster name** and the provided **BackupPrefix**.
+- **Credentials:** This is a `SecretReference` located in the HostedCluster namespace using the same name as the backup plus the "credentials" suffix.
+- **ServiceConfigType:** This is a enum field which will reference the remote cloud providers to upload the backup object (e.g s3, azure, odf).
+- **S3ServiceConfig:** This field is a new type dedicated to the S3 provider. It will contain the details about the remote destination you will upload that backup (warning: This field name is subject to change)
+- **AzureServiceConfig:** This field is a new type dedicated to the Azure provider. The implementation details will be included in the future (warning: This field name is subject to change)
+- **ODFServiceConfig:** This field is a new type dedicated to the ODF provider. The implementation details will be included in the future (warning: This field name is subject to change)
+
+Each provider struct will be different and for the initial development we will focus on S3 because it could cover AWS and on-premises (MinIO) storage.
+
+- `s3ServiceConfig`:
+  - **ServiceURL:** Points to the URL necessary to upload the object.
+  - **Region:** This will set the Region for the destination bucket.
+  - **Tags:** Necessary tags to label the backup in he destination resources.
+
+### Workflow Description
+
+#### Backup
+
+When a user with a HostedCluster intends to perform a backup, the initial step involves modifying the HostedCluster by introducing a new field within the etcd API field. This new field will serve to incorporate the required parameters and configurations essential for executing backup operations seamlessly within the HostedCluster environment.
+
+```yaml
+...
+spec:
+  controllerAvailabilityPolicy: SingleReplica
+  dns:
+    baseDomain: hypershiftbm.lab
+  etcd:
+    managed:
+      storage:
+        persistentVolume:
+          size: 8Gi
+        restoreSnapshotURL: null
+        type: PersistentVolume
+    managementType: Managed
+    backup:
+      etcdBackup:
+        timeZone: "UTC"
```

Comment:
Why do we want to expose a timezone here? If we always expect the crontab to be in UTC we sufficiently expose every possible schedule, right? 

It is not clear to me what the time zone field does when the schedule is something like `@weekly` as in the current example. What would a user expect when they have `@weekly` and change the timezone from CET to PST?

Reply:
disregard, we are just embedding the OCP `config.openshift.io/v1alpha1:Backup` resource.